presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -948,6 +949,10 @@ public OptionalLong metadataDelete(ConnectorSession session, ConnectorTableHandl
             throw new TableNotFoundException(handle.getSchemaTableName());
         }
 
+        if (!table.get().getTableType().equals(MANAGED_TABLE.toString())) {
```

Comment:
@dain and I were just discussing this. We want to allow dropping partitions from an external table (it's a metadata-only operation). This check should only happen for the non-partitioned case, since that actually deletes delete.